activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -86,6 +86,20 @@ def supports_lazy_transactions?
         true
       end
 
+      def column_definitions(table_name)
+        if !mariadb? && get_database_version > "8.0.2"
+          execute_and_free("SHOW FULL FIELDS FROM #{quote_table_name(table_name)}", "SCHEMA") do |result|
+            each_hash(result).each do |col_info|
+              if col_info[:Type].start_with?("varbinary") && col_info.fetch(:Default)&.start_with?("0x")
```

Comment:
Does this affect the `binary` type as well?

Reply:
Yes! I added a test for the binary type as well.